src/main/java/io/scim2/swagger/client/ScimApiClient.java

```diff
@@ -536,39 +610,98 @@ public boolean isDebugging() {
      * @return ScimApiClient
      */
     public ScimApiClient setDebugging(boolean debugging) {
-        if (debugging != this.debugging) {
-            if (debugging) {
-                loggingInterceptor = new HttpLoggingInterceptor();
-                loggingInterceptor.setLevel(Level.BODY);
-                httpClient.interceptors().add(loggingInterceptor);
-            } else {
-                httpClient.interceptors().remove(loggingInterceptor);
-                loggingInterceptor = null;
+
+        try {
+            if (debugging != this.debugging) {
+                if (debugging) {
+                    logger.info("Enabling debug mode");
+
+                    // Create the logging interceptors.
+                    HttpRequestInterceptor requestInterceptor =
+                            (request, context) -> logger.debug(getRequestInfo(request));
+                    HttpResponseInterceptor responseInterceptor =
+                            (response, context) -> logger.debug(getResponseInfo(response));
+
+                    httpClient = HttpClients.custom()
+                            .addInterceptorFirst(requestInterceptor)
+                            .addInterceptorFirst(responseInterceptor)
+                            .build();
+                } else {
+                    logger.info("Disabling debug mode");
+                    ((CloseableHttpClient) httpClient).close();
+                    httpClient = httpClientBuilder.build();
+                }
             }
+        } catch (IOException e) {
+            logger.error("An IOException occurred: {}", e);
         }
         this.debugging = debugging;
         return this;
     }
 
+    private String getRequestInfo(HttpRequest request) throws IOException {
+
+        String message = "\nRequest - "
+                + request.getRequestLine().getMethod() + " "
+                + request.getRequestLine().getUri()
+                + "\nHeaders: ["
+                + Arrays.stream(request.getAllHeaders())
+                .map(header -> header.getName() + ": " + header.getValue())
+                .collect(Collectors.joining(", "))
+                + "]";
```

Comment:
we can extract this out to a separate method. to improve readability

private String getHeadersInfo(HttpRequest request) {
    return Arrays.stream(request.getAllHeaders())
            .map(header -> header.getName() + ": " + header.getValue())
            .collect(Collectors.joining(", "));
}